eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php

```diff
@@ -50,6 +52,10 @@ public function downloadBinaryFileAction($contentId, $fieldIdentifier, $filename
             $content = $this->contentService->loadContent($contentId);
         }
 
+        if ($content->contentInfo->isTrashed()) {
+            throw new NotFoundException('File', null);
```

Comment:
Maybe pass `$filename` here instead of null?

Reply:
@ViniTou wouldn't it be a bit of a security breach? I mean, hiding the name of the file was kinda intentional here.

Reply:
The filename is still in the url tho, I am right?

Reply:
Ah yes, indeed, I'll add `$filename`, then